camel/typing.py

```diff
@@ -49,9 +72,33 @@ def token_limit(self) -> int:
             return 32768
         elif self is ModelType.STUB:
             return 4096
+        elif self is ModelType.LLAMA_2:
+            return 4096
+        elif self is ModelType.VICUNA:
+            # reference: https://lmsys.org/blog/2023-03-30-vicuna/
+            return 2048
+        elif self is ModelType.VICUNA_16K:
+            return 16384
         else:
             raise ValueError("Unknown model type")
 
+    def match_model(self, model_name: str) -> bool:
+        r"""Checks whether the model type and the model name matches.
+
+        Args:
+            model_name (str): The name of the model, e.g. "vicuna-7b-v1.5".
+        Returns:
+            bool: Whether the model type mathches the model name.
+        """
+        if self is ModelType.VICUNA:
+            pattern = r'^vicuna-\d+b-v\d+\.\d+$'
+            return bool(re.match(pattern, model_name))
+        elif self is ModelType.VICUNA_16K:
+            pattern = r'^vicuna-\d+b-v\d+\.\d+-16k$'
+            return bool(re.match(pattern, model_name))
+        else:
+            return self.value in model_name.lower()
```

Comment:
This is because there are different versions and naming styles of the same model. For example, `Llama-2-7b-chat-hf` should have type `LLAMA_2` but its name is not exactly `llama-2`. The model name will be pased to the server for checking so we should not arbitrarily assign it.

Reply:
A regex would be more appropriate here. 'in' can make mistakes.

Reply:
@Obs01ete But I am afraid that for LLaMA-2-based models, `in` might be sufficient for checking this match? I searched LLaMA-2-based models in HuggingFace and the names of them all contain "llama-2" or "llama2".

Reply:
What if we have a local "camel-model-based-on-llama"?

Reply:
Got it. I am good with this solution.